src/detect.rs

```diff
@@ -76,6 +70,40 @@ mod tests {
             ))),
             Err(EnvError::Missing)
         );
+    }
+
+    #[test]
+    #[cfg(target_os = "linux")]
+    fn host_detect_success() {
+        assert!(
+            pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7-preview"))).is_ok()
+        );
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.2"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.3"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.4"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.5"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.6"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.7"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.8"))).is_ok());
+        assert!(pwsh_host_detect(Some(OsString::from("/opt/microsoft/powershell/7.9"))).is_ok());
```

Comment:
You should also make sure stuff that like `/opt/microsoft/powershell/7` is working.

> The /usr/local/microsoft/powershell/6 folder is replaced by /usr/local/microsoft/powershell/7.

[Source](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux?view=powershell-7.1)